rust/private/rust.bzl

```
@@ -1139,6 +1139,82 @@ rust_test_binary = rule(
         """),
 )
 
+def rust_test_suite(name, srcs, tags = [], **kwargs):
```

```suggestion
def cargo_test_suite(name, srcs, **kwargs):
```
Since this is more of a cargo pattern, would you be alright with going with this name instead?

(I added in additional changes based on my comment on line 1215)

---

I don't think it's appropriate to prefix this with `cargo` since cargo is not being used in any way here. The design does come from cargo but I feel it's an effective pattern particularly for Rust and it's worth continuing in Bazel.

---

@illicitonion do you have opinions here?

This currently my only hang-up on the PR, but my opinions aren't too strong on this.

The main thing, in my head is that it's a cargo-style pattern, and that's probably the easiest connection for folks to make, and that `rust_test_suite` seems a little too general for this type of pattern.

---

I feel pretty strongly that this shouldn't be `cargo_test_suite`. I think that's pretty misleading. Another name would be fine but I also think `rust_test_suite` is totally appropriate.

---

Given that I haven't seen this sort of test pattern used outside of cargo, and we already have cargo_build_script, I'm not sure I understand the strong opposition.

Are there public examples of where this would be use in a context of a project that isn't also building with Cargo?

---

I do not have public examples. The difference between this test suite and cargo_build_script is that build.rs is a concept invented by cargo that would never have naturally occurred while integration tests are. Do you have any rust projects with integration tests? How did you implement those?

---

tl;dr: `rust_test_suite` works for me.

I would definitely feel strange about naming this `cargo_test_suite`, as there nothing cargo-y about it other than that it happens to be useful if you happen to use the cargo file layout - it would equally apply without the `tests` dir, e.g.
```python
rust_test_suite(
    name = "tests",
    srcs = ["*_test.rs"],
)
```

The `${language}_test` -> `${language}_test_suite` convention is pretty common (e.g. [`scala_test_suite`](https://github.com/bazelbuild/rules_scala/blob/master/docs/scala_test_suite.md) for `scala_test`. Apparently `java_test_suite` didn't make it from `blaze` to `bazel`, but my company certainly has a `java_test_suite` macro which looks a lot like this does but for Java)...